### DISCUSSION TITLE
[Feature request] Auto temperature as in Creality Print v5.

### DISCUSSION THREAD

**Discussion by SPYKE357** [OrcaSlicer/OrcaSlicer#5361] [General]
Hi 

would it be possible to implement the Flow Temperature Graph in the Auto Temperature selection? Seemingly helps with the benchy hull line/perimeter extrusion after solid infill.

**Comment by SPYKE357** [comment]
Copied my Orca config to CrealityPv5 to test the auto temp feature further. really helpful with matte and silk filaments if the max volumetric flow and temps have been tested. gives a more pleasing even finish on the part especially at higher print speeds.

for example, i have a small simplified dinosaur model that has been printed in both matte and silk filaments. on my usual presets, the lower speed/flow areas would be much shinier than areas of the model printed at higher speeds/flow rates if the temp is constant, 220C in this scenario. this gave me the idea to get a more uniform and intentional finish i would either have to play with temps at the different layers, or adjust extrusion multipliers etc with layer modifiers. 

I tested CrealityPv5's auto temp (adjusted the temps and flows in the graph section as required per filament) and achieved uniformity in the finish of the matte/silk prints. i can purposefully get silk to look more matte/silky throughout the whole part or vice versa, by using data from the temps and max volumetric flow test from CNCKitchen, i can see if my filament was slightly under extruding or over extruding to get a specific effect and adjust the temps vs flows in the CrealityPv5 accordingly. been getting fantastic results

I've only tested this feature for looks but i can certainly imagine it helping with strength on specific prints, layer to layer adhesion in areas of prints where underextrusion may be experienced due to changes in flow. ABS/PETG would certainly come to mind here that could certainly benefit from this. also areas with lower flow would kick in a lower temperature so could also be benificial for parts that have overhangs over a layer range. (it certainly helped with overhangs on the model i was testing)

Please add this feature. it really is awesome. An amazing feature to really dial a certain look. Currently using Orca but copying my configs to CrealityPv5 for silk and matte filaments only.

**Comment by LuccoJ** [comment]
Creality Print 4 also had automatic temperature, although it may have been implemented differently as it's based on a different engine.
What I seem to find is that it helps with bridging and overhangs, for example I printed a small flute on Orca and it's practically not playable, while with Creality Print 4 it plays fine, and visibly, the differences involve much less floppy bridges.

**Comment by sb53systems** [comment]
Hi guys, you can use my post processing script (Gcode flow and temperature controller).  
https://github.com/sb53systems/G-Code-Flow-Temperature-Controller. 
Have fun and happy 3d printing )
